### ISSUE TITLE
Behavior of `createstub` and `__all__`

### ISSUE THREAD

**Issue by steverice** [microsoft/pyright#2391]
**Is your feature request related to a problem? Please describe.**
Since [pyright 1.1.168](https://github.com/microsoft/pyright/releases/tag/1.1.168), symbols that are imported into stub files but not explicitly exported using `import x as x` syntax are reported as not present in that module.

I understand that this follows the [behavior described in PEP 484](https://www.python.org/dev/peps/pep-0484/#stub-files):
> Modules and variables imported into the stub are not considered exported from the stub unless the import uses the `import ... as ...` form or the equivalent `from ... import ... as ...` form. (UPDATE: To clarify, the intention here is that only names imported using the form `X as X` will be exported, i.e. the name before and after as must be the same.)

Many libraries I'm using do not follow this pattern, but instead declare their intent to export symbols using `__all__`, for example:
```python
import other_symbol
__all__ = ['other_symbol']
```

When creating stubs for these libraries with `pyright --createstub`, the stub files do not include the `__all__` declaration, nor are the `import` statements modified. As a result, a symbol which is importable from the library module at runtime is flagged as an "unknown import symbol" by pyright despite not being reflective of a true runtime import error.

**Describe the solution you'd like**
My suggestion is that when pyright creates stubs with `--createstub` (which I do often, having a number of esoteric/untyped libraries), it should copy the `__all__` declaration from modules into their respective stubs.

Caveat is I don't know exactly what the "right" solution is here, and I think that adhering to PEP 484 semantics is a worthy goal. Were I to manual write type stubs for a module, I would probably modify the imports in the stub to follow the `X as X` form.

However, when generating stubs automatically it seems less error-prone to simply copy over the `__all__` declaration from the module into the stub. This also seems in line with pyright's [guidance for defining library interfaces](https://github.com/microsoft/pyright/blob/d5878534fc467fc70a5b5ac13b8a09c496ffd22e/docs/typed-libraries.md#library-interface) which supports `__all__` as a way of describing the same semantics.

**Additional context**
IMO there is a distinction to be made between "supported" type stubs i.e. those provided by typeshed or a `py.typed` package and "shim" stubs created by `--createstub` as a best effort. Therefore I think it is reasonable to require PEP 484 rules for "supported" stubs while not applying those same rules to the module code that is being used to generate "shim" stubs.

I think the solution I've proposed is a good compromise because it does not alter the type checker's behavior, but only the output of the stub generator. The resulting stub more closely mimics the way its corresponding code behaves.

**Comment by erictraut** [comment]
This is included in pyright 1.1.176, which I just published. It will also be included in the next release of pylance.
